frontend/src/App.js

```diff
@@ -1,21 +1,56 @@
 import "./App.css"
 import Login from "./components/Login/Login"
-import { BrowserRouter as Router, Routes, Route } from "react-router-dom"
+import { BrowserRouter, useNavigate, Routes, Route } from "react-router-dom"
 import Signup from "./components/Signup/Signup"
 import "react-toastify/dist/ReactToastify.css"
 import { ToastContainer } from "react-toastify"
+import Header from "./components/Header/Header"
+import { reducer, initialState } from "./reducers/userReducer"
+import React, { useEffect, createContext, useReducer, useContext } from "react"
 
-function App() {
+export const UserContext = createContext()
+
+const Routing = () => {
+  const navigate = useNavigate()
+  const { state, dispatch } = useContext(UserContext)
+
+  useEffect(() => {
+    const user = JSON.parse(localStorage.getItem("user"))
+
+    if (user) {
+      dispatch({ type: "USER", payload: user })
+      navigate("/")
+    } else {
+      navigate("/register")
+    }
+  }, [])
   return (
-    <div className="App">
-      <Router>
-        <Routes>
-          <Route path="/login" element={<Login />}></Route>
-          <Route path="/register" element={<Signup />}></Route>
-        </Routes>
-      </Router>
+    <>
+      <Routes>
+        <Route path="/"></Route>
+        <Route path="/login" element={<Login />}></Route>
+        <Route path="/register" element={<Signup />}></Route>
+      </Routes>
       <ToastContainer />
-    </div>
+    </>
+  )
+}
+
+function App() {
+  const [state, dispatch] = useReducer(reducer, initialState)
+  return (
+    <UserContext.Provider value={{ state, dispatch }}>
+      <BrowserRouter
+        className="App"
+        style={{
```

Comment:
Style and className property can only be applied to HTML tags and Browser Router is not a HTML tag.